tests/test-mongoc-uri.c

```diff
@@ -96,12 +98,15 @@ test_mongoc_uri_new (void)
    ASSERT_CMPSTR(bson_iter_utf8(&iter, NULL), "1");
    mongoc_uri_destroy(uri);
 
-   uri = mongoc_uri_new("mongodb://localhost:27017/?readPreferenceTags=dc:ny&readPreferenceTags=");
+   uri = mongoc_uri_new("mongodb://localhost:27017/?readPreference=secondaryPreferred&readPreferenceTags=dc:ny&readPreferenceTags=");
```

Comment:
Previously, such a URI would result in the client's read preference having tags with the primary mode, which is invalid according to the read preference spec and `mongoc_read_prefs_is_valid()`. That said, `mongoc_read_prefs_is_valid()` was never actually called anywhere, so I'm not sure how libmongoc handled such a structure.